script/tool/lib/src/common.dart

```diff
@@ -466,3 +542,69 @@ class ProcessRunner {
     return 'ERROR: Unable to execute "$executable ${args.join(' ')}"$workdir.';
   }
 }
+
+/// Finding diffs based on `baseGitDir` and `baseSha`.
+class GitVersionFinder {
+  /// Constructor
+  GitVersionFinder(this.baseGitDir, this.baseSha);
+
+  /// The top level directory of the git repo.
+  ///
+  /// That is where the .git/ folder exists.
+  final GitDir baseGitDir;
+
+  /// The base sha used to get diff.
+  final String baseSha;
+
+  static bool _isPubspec(String file) {
+    return file.trim().endsWith('pubspec.yaml');
+  }
+
+  /// Get a list of all the pubspec.yaml file that is changed.
+  Future<List<String>> getChangedPubSpecs() async {
+    return (await getChangedFiles()).where(_isPubspec).toList();
+  }
+
+  /// Get a list of all the changed files.
+  Future<List<String>> getChangedFiles() async {
+    final String baseSha = await _getBaseSha();
+    final io.ProcessResult changedFilesCommand = await baseGitDir
+        .runCommand(<String>['diff', '--name-only', '$baseSha', 'HEAD']);
+    print('Determine diff with base sha: $baseSha');
+    if (changedFilesCommand.stdout.toString() == null ||
+        changedFilesCommand.stdout.toString().isEmpty) {
+      return <String>[];
+    }
+    final List<String> changedFiles = changedFilesCommand.stdout
```

Comment:
`changedFilesCommand.stdout.toString()` is used three times; let's make it a local variable. (You could avoid the early return and its two checks by just setting that string with an `?? ''` and then proceeding; an empty string fed to the logic below should give you an empty list).

Reply:
nice, done